modules/json/src/smithy4s/json/internals/SchemaVisitorJCodec.scala

```diff
@@ -394,11 +394,8 @@ private[smithy4s] class SchemaVisitorJCodec(
       }
 
       def encodeValue(x: Timestamp, out: JsonWriter): Unit = {
-        if (x.nano == 0) {
-          out.writeVal(x.epochSecond)
-        } else {
-          out.writeVal(BigDecimal(x.epochSecond) + x.nano / 1000000000.0)
-        }
+        val bd = BigDecimal(x.epochSecond) + (x.nano * BigDecimal(10).pow(-9))
```

Comment:
**issue:** shouldn't it be `BigDecimal(x.epochSecond) + (BigDecimal(x.nano) * BigDecimal(10).pow(-9))`

otherwise we might lose some information due to overflow before the conversion to `BigDecimal` happens.

Reply:
The result of this multiplication is already a `BigDecimal`, so I think it is fine